docs/reference/developer/new-plotting-examples.ipynb

```diff
@@ -0,0 +1,579 @@
+{
```

Comment:
Instead of a notebook, should we write these as tests directly?

Reply:
This is not really the kind of tests I want to write first. I would like to write tests for the individual classes (the graph, the view, the mesh etc...). I think we would write high-level tests with typical examples once we have a wrapper function that would serve the purpose of the current `plot()` function?

Reply:
- If this is not part of the public (if low-level) API then I don't think we should put a notebook into the docs folder.
- If it is public (low-level) API, then it is part of a contract and should be tested.

Testing individual classes may be counter-productive? Remember: Test behaviors, not methods.

Reply:
Sure I want to test if I e.g. add a new view to a graph node, I get the plot i'm expecting.
But right now these examples are just giving a preview of what we can do. The API isn't final, it's just for our internal use.
So I thought we should keep it somewhere, but maybe we should just have it under version control but not in the docs.

Reply:
If you are going to refactor the examples with API changes, then what difference does it make whether they are tests or in a notebook? Aren't you risking wasting your work if you put things in a notebook? I often feel that I make this mistake myself: Debugging/testing something in a notebook, where I should have written it as a test from the beginning (or moved it into one as soon as I got it working).